src/components/clients/ClientForm.js

```diff
@@ -36,16 +67,43 @@ class ClientForm extends React.Component {
   }
 
   render() {
+    let authRedirectUrls = [];
+    for (let i = 0; i < this.state.authRedirectUrlsCount; i++) {
+      authRedirectUrls.push(
+        <input value={this.state.authRedirectUrls[i]} onFocus={this.onFocus}
+               onChange={this.onRedirectUrlChange} id={i} key={i} type="text"
+               name="authRedirectUrls" className="form-control"/>
+      );
+    }
+
+    let authFailureRedirectUrls = [];
+    for (let i = 0; i < this.state.authFailureRedirectUrlsCount; i++) {
+      authFailureRedirectUrls.push(
+        <input value={this.state.authFailureRedirectUrls[i]}
+               onFocus={this.onFocus}
+               onChange={this.onRedirectUrlChange} id={i} key={i} type="text"
+               name="authFailureRedirectUrls" className="form-control"/>
+      );
+    }
+
     return (
       <form className="container" onSubmit={this.onSubmit}>
         <div className="form-group">
           <label className="control-label">Client name</label>
-          <input
-            value={this.state.clientName}
-            onChange={this.onChange}
-            type="text"
-            name="clientName"
-            className="form-control"/>
+          <input value={this.state.clientName} onChange={this.onChange}
+                 type="text" name="clientName" className="form-control"/>
+          <label className="control-label">Auth redirect URLs</label>
```

Comment:
I think you really want a fieldset here. This would help bring some structure to the form.
You'd be able to add the onChange event handler directly on the fieldset too.